code/API_definitions/qod-api.yaml

```diff
@@ -452,43 +437,112 @@ components:
       content:
         application/json:
           schema:
-            $ref: "#/components/schemas/ErrorInfo"
+            allOf: 
+              - $ref: "#/components/schemas/ErrorInfo"
+              - type: object
+                properties: 
+                  status:
+                    type: string
+                    enum: ["400"]
+                  code:
+                    type: string
+                    enum: ["INVALID_INPUT"]
```

Comment:
My understanding is that the point of this `allOf` specialisation is to link the `status` (in this case, `"400"`) with the `code` (`"INVALID_INPUT"`). I don't see the value in this, and would prefer just to give a comprehensive list of examples, differing only by `message` for each of the response schema (`Generic400`, etc.).

If the status and code of a given response received by the client do not align (e.g. `status` is `"404"` and `code` is `"INVALID_INPUT"`), I don't think the client would need this specialisation to detect that the server implementation was broken. So my preference would be simplify life for the client developer reading the API specification and have all error responses being simple `ErrorInfo` objects with no specialisation but a good list of examples they might receive.

Reply:
Given that this change is triggered from [Api design guidelines](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/API-design-guidelines.md) would it be better if we extract these common error messages and add it to commonalities repo - we are moving in that direction looking [here](https://github.com/camaraproject/WorkingGroups/blob/main/Commonalities/documentation/CAMARA_common.json)? We could later reference the same in QoD spec. Of course there will be more open items like versioning spec across repos.

The discussion will be wider and the `common` elements will be binding on all sub-projects to include it in their respective specs. My humble opinion!

Reply:
Thanks @patrice-conil 

On your specific point:

> If status and code are so related...why do we need both?

There are two reasons:
- HTTP status codes are a standard (primarily IETF RFC 9110) that is well known to developers, whereas CAMARA error codes (`"INVALID_INPUT"`, etc.) are not a standard (albeit that they are fairly intuitive)
- There is not necessarily a one-to-one relationship between `status` and `code`. For example, I think that both `"INVALID_INPUT"` and `"OUT_OF_RANGE"` are `"400"` errors.

So both are useful.

Reply:
Thanks @eric-murray,
Good point.
Everyday I learn from my mistakes ... and I learn a lot. :D

Reply:
I agree with @SfnUser that this is something to take to Commonalities, but probably we are leading the path here.

Regarding whether responses must be restrictive or not, it depends on how far we want to go with OAS3 capabilities.

If we just ask error responses just to comply with ErrorModel, a server returning

```
HTTP 400
{
  "status": "HTTP 400",
  "code": "MY_OWN_CODE",
  "message": "My message"
}
```
would strictly be API compliant. We can encourage implementations to follow guidelines and examples without defining specific schemas per response, and it is true that some tools have problems dealing with complex schemas for auto generation of data models, so there are pros and cons. In the end we have to assure that all implementations follow the same model.

Reply:
Thanks @eric-murray, 
It's clear now.

Reply:
Guess that's done for now